File: EnBotJsAPI/ServerFunctions/CreateButtons.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Text.RegularExpressions;

namespace EnBot.EnBotJsAPI {
    public class CreateButtons : ServerFunctionAPI {
        public override string URN => "create-buttons";
        public new class Response : ServerFunctionAPI.Response {
            [JsonProperty("channel")]
            public string Channel { get; private set; }
            [JsonProperty("content")]
            public string Content { get; private set; }
            [JsonProperty("buttons")]
            public List<Button> Buttons { get; private set; }
            public class Button : Inner {
                [JsonProperty("text")]
                public string Text { get; private set; }
                [JsonProperty("color")]
                public string Color { get; private set; }
                public static class Colors {
                    public static readonly string Blurple = "blurple";
                    public static readonly string Grey = "grey";
                    public static readonly string Green = "green";
                    public static readonly string Red = "red";
                }
                public Button(string text, string color) {
                    Text = text;
                    Color = color;
                }
                public override string Validate() {
                    // Text
                    if (!Regex.IsMatch(Text, @"^.{1,64}$"))
                        return "Buttons.Text name is not valid";
                    // Color
                    if (!Regex.IsMatch(Color, @"^(blurple|grey|green|red)$"))
                        return "Buttons.Color name is not valid";
                    return null;
                }
            }
            public Response(string channel, string content, List<Button> buttons) {
                Channel = channel;
                Content = content;
                Buttons = buttons;
            }
            public override string Validate() {
                string error = null;
                // Channel
                // Buttons
                foreach (var button in Buttons)
                    if ((error = button.Validate()) != null)
                        return error;
                return error;
            }
        }
    }
}

File: EnBotJsHelper/server/routes/create-buttons.ts
import Server from '@server/Server';
import Express, { NextFunction, Response, Request } from 'express';
import asyncHandler from 'express-async-handler';
import { CreateButtonsAPI, verification } from '@api/create-buttons';
import validate from '@utils/validate';
import Discord from 'discord.js';
import 'colors';
import { MessageActionRow, MessageButton } from 'discord-buttons';

const urnRoot = "/create-buttons";

export default function(app: Express.Application): void {
    var handler = app.get('handler') as typeof asyncHandler;
    app.post(urnRoot, handler(async (req: Request, res: Response, next: NextFunction) => {
        // If response already sent
        if (Server.runtimeInfo[req.id].isSending) {
            next(); return;
        }
        // Body validation
        let reqBody = req.body as CreateButtonsAPI['req'];
        if (!validate(reqBody, verification)) {
            res.sendStatus(400/* Bad request */);
            Server.runtimeInfo[req.id].isSending = true;
            next(); return;
        }
        // Discord buttons creating
        var discordClient = app.get('discordClient') as Discord.Client;
        //var channel = discordClient.channels.cache.get(reqBody.channel.toString());
        //if (!channel)
        var channel = await discordClient.channels.fetch(reqBody.channel.toString());
        if (!channel) {
            res.sendStatus(404/* Not found */);
            Server.runtimeInfo[req.id].isSending = true;
            next(); return;
        }
        var readyButtons : Array<MessageButton> = [];
        var discordButton = app.get('discordButton');
        reqBody.buttons.forEach((button, i) => readyButtons.push(
            new (discordButton as any)
                .MessageButton()
                .setLabel(button.text)
                .setStyle(button.color) // blurple, grey, green, red
                .setID(i)
            )
        );
        var buttonsRow = new MessageActionRow().addComponents(...readyButtons);
        (channel as any).send(reqBody.content, { components: buttonsRow });
        // Sending responce
        res.sendStatus(200/* OK */);
        Server.runtimeInfo[req.id].isSending = true; next();
    }));
}